File: app/src/main/java/com/marked/pixsee/signup/SignUpEmailFragment.java
package com.marked.pixsee.signup;


import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.util.Patterns;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.marked.pixsee.R;
import com.marked.pixsee.utility.Utils;

/**
 * A simple [Fragment] subclass.
 * Activities that contain this fragment must implement the
 * [SignUpEmailFragmentInteraction] interface
 * to handle interaction events.
 * Use the [SignUpEmailFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
public class SignUpEmailFragment extends Fragment {
	private SignUpEmailFragmentInteraction mListener;
	private TextView emailEditText;
	private TextView hiUser;
	private Button nextButton;

	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		View rootView = inflater.inflate(R.layout.fragment_sign_up_email, container, false);
		hiUser = (TextView) rootView.findViewById(R.id.hiUser);
		emailEditText = (TextView) rootView.findViewById(R.id.emailEditText);

		hiUser.setText("Beautiful name " + getArguments().getString(NAME_TAG));
		nextButton = (Button) rootView.findViewById(R.id.nextButton);
		nextButton.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				if (!Utils.isOnline(getActivity())) {
					Utils.showNoConnectionDialog(getActivity());
				} else {
					String email = emailEditText.getText().toString().trim();
					if (email.isEmpty())
						Toast.makeText(getActivity(), "Please enter an email", Toast.LENGTH_SHORT).show();
					else if (Patterns.EMAIL_ADDRESS.matcher(email).matches())
						onNextPressed(email);
					else
						Toast.makeText(getActivity(), "Invalid email adress", Toast.LENGTH_SHORT).show();
				}
			}
		});

		// Inflate the layout for this fragment
		return rootView;
	}


	// TODO: Rename method, update argument and hook method into UI event
	public void onNextPressed(String emailEditText) {
		if (mListener != null) {
			mListener.onSaveEmail(emailEditText);
		}
	}

	@Override
	public void onAttach(Context context) {
		super.onAttach(context);
		if (context instanceof SignUpEmailFragmentInteraction)
			mListener = (SignUpEmailFragmentInteraction) context;
		else
			throw new RuntimeException(context.toString() + " must implement SignUpEmailFragmentInteraction");
	}

	@Override
	public void onDetach() {
		super.onDetach();
		mListener = null;
	}

	interface SignUpEmailFragmentInteraction {
		// TODO: Update argument type and name
		void onSaveEmail(String emailEditText);
	}

	public static final String NAME_TAG = "NAME";

	public static SignUpEmailFragment newInstance(String name) {
		Bundle bundle = new Bundle();
		bundle.putString(NAME_TAG, name);
		SignUpEmailFragment fragment = new SignUpEmailFragment();
		fragment.setArguments(bundle);
		return fragment;
	}
}// Required empty public constructor

File: app/src/main/java/com/marked/pixsee/signup/SignUpEmailFragment.kt
package com.marked.pixsee.signup

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Patterns
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.marked.pixsee.R
import com.marked.pixsee.utility.Utils
import kotlinx.android.synthetic.main.fragment_sign_up_email.*
import kotlinx.android.synthetic.main.fragment_sign_up_email.view.*
import org.jetbrains.anko.onClick
import org.jetbrains.anko.toast
import java.io.UnsupportedEncodingException
/**
 * A simple [Fragment] subclass.
 * Activities that contain this fragment must implement the
 * [SignUpEmailFragmentInteraction] interface
 * to handle interaction events.
 * Use the [SignUpEmailFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class SignUpEmailFragment : Fragment() {
	private var mListener: SignUpEmailFragmentInteraction? = null

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		val rootView = inflater.inflate(R.layout.fragment_sign_up_email, container, false)
		rootView.hiUser.text = "Beautiful name ${arguments.getString(Companion.NAME_TAG)},"
		rootView.nextButton.onClick {
			if (!Utils.isOnline(activity)) {
				Utils.showNoConnectionDialog(activity)
			} else {
				val email = emailEditText.text.toString().trim { it <= ' ' }
				if (email.isNullOrBlank())
					activity.toast("Please enter an email")
				else if (Patterns.EMAIL_ADDRESS.matcher(email).matches())
					onNextPressed(email)
				else
					activity.toast("Invalid email adress")
			}
		}
		// Inflate the layout for this fragment
		return rootView
	}

	// TODO: Rename method, update argument and hook method into UI event
	fun onNextPressed(email: String) {
		if (mListener != null) {
			try {
				mListener!!.onSaveEmail(email)
			} catch (e: UnsupportedEncodingException) {
				e.printStackTrace()
			}

		}
	}

	override fun onAttach(context: Context?) {
		super.onAttach(context)
		if (context is SignUpEmailFragmentInteraction)
			mListener = context as SignUpEmailFragmentInteraction?
		else
			throw RuntimeException(context!!.toString() + " must implement SignUpEmailFragmentInteraction")
	}

	override fun onDetach() {
		super.onDetach()
		mListener = null
	}

	interface SignUpEmailFragmentInteraction {
		// TODO: Update argument type and name
		fun onSaveEmail(email: String)
	}

	companion object {
		val NAME_TAG = "NAME"
		fun newInstance(name: String): SignUpEmailFragment {
			val bundle = Bundle()
			bundle.putString(Companion.NAME_TAG, name)
			val fragment = SignUpEmailFragment()
			fragment.arguments = bundle
			return fragment
		}
	}
}// Required empty public constructor
